Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.sb.in

```diff
@@ -22,8 +22,308 @@
+#if !ENABLE(CFPREFS_DIRECT_MODE)
+    (global-name "com.apple.cfprefsd.daemon")
+#endif
+    (global-name "com.apple.logd")
+    (global-name "com.apple.logd.events")
+)
+
+(deny mach-lookup (with no-log)
+    (global-name "com.apple.CoreServices.coreservicesd")
+    (global-name "com.apple.DiskArbitration.diskarbitrationd")
+    (global-name "com.apple.ViewBridgeAuxiliary")
+    (global-name "com.apple.system.opendirectoryd.membership"))
+
+;; CoreFoundation. We don't import com.apple.corefoundation.sb, because it allows unnecessary access to pasteboard.
+(allow mach-lookup
+    (global-name-prefix "com.apple.distributed_notifications")
+#if !HAVE(CSCHECKFIXDISABLE)
+    (global-name "com.apple.CoreServices.coreservicesd")
+#endif
+)
+
+(deny file-write-create (vnode-type SYMLINK))
+
+;; Reserve a namespace for additional protected extended attributes.
+(deny file-read-xattr file-write-xattr (xattr-prefix "com.apple.security.private."))
+
+;; Analytics.
+(allow mach-lookup (global-name "com.apple.analyticsd"))
+(allow mach-lookup (global-name "com.apple.awdd"))
+
+;; Daemon prefs.
+(allow user-preference-read user-preference-write
+    (preference-domain "com.apple.webkit.webpushd"))
+
+;; Push notification registration.
+(allow mach-lookup
+    (global-name "com.apple.apsd")
+    (global-name "com.apple.windowserver.active")) ;; for -[APSSystemUser isLoggedIn]
+
+;; Allow webpushd to launch UI processes in response to a push via LaunchServices.
+(allow lsopen)
+
+(allow mach-lookup
+    (global-name "com.apple.coreservices.launchservicesd")
+    (global-name "com.apple.coreservices.quarantine-resolver")
+    (global-name "com.apple.lsd.mapdb"))
+
+(allow user-preference-read
+    (preference-domain
+         "com.apple.inputmethodkit"
+         "com.apple.inputsources"
+         "com.apple.LaunchServices"
+         "kCFPreferencesAnyApplication"))
+
+;; We might need to create ~/Library/WebKit and ~/Library/WebKit/WebPush on a clean install.
+(allow file-write-create
+    (require-all (vnode-type DIRECTORY) (home-literal "/Library/WebKit"))
+    (require-all (vnode-type DIRECTORY) (home-literal "/Library/WebKit/WebPush")))
+
+;; Push database.
+(allow file*
+    (home-literal "/Library/WebKit/WebPush/PushDatabase.db")
+    (home-literal "/Library/WebKit/WebPush/PushDatabase.db-shm")
+    (home-literal "/Library/WebKit/WebPush/PushDatabase.db-wal")
+    (home-literal "/Library/WebKit/WebPush/PushDatabase.db-lock")
+    (home-literal "/Library/WebKit/WebPush/PushDatabase.db-journal"))
+
+;; Let SQLite open the parent directory of the database for fsync purposes.
+(allow file-read-data (require-all (vnode-type DIRECTORY) (home-prefix "/Library/WebKit/WebPush")))
+
+;; SQLite might spill large transactions to TMPDIR.
+(allow file-write*
+    (subpath (param "DARWIN_USER_TEMP_DIR"))
+    (subpath (param "DARWIN_USER_CACHE_DIR")))
+
+(allow file-read* file-write*
+    (mount-relative-regex #"^/\.TemporaryItems/"))
```

Comment:
If possible, I think it would be good to avoid (write) access to this directory.

Reply:
You can also add (with telemetry) here in order to determine if it is needed.

Reply:
Looks like Foundation uses it for atomic writes, which we don't use explicitly but could be used by frameworks that we call. I think keeping the line with telemetry is a good idea for now.